### PR TITLE
Try to stop full URLs as gameboard IDs

### PR DESCRIPTION
**Pull request by bh412**
Simple check for "http" being manually entered into the gameboard ID if provided by a staff member. Won't affect http being present in random generated ID from backend if no ID provided by staff.

### CODE REVIEW

**Review by jsharkey13 — CHANGES_REQUESTED** [review]
This still allows invalid IDs, and blocks valid ones. Just blocking anything not matching something like `^[a-z0-9_-]+$` would be better, I think. Not sure if we should allow uppercase letters, there seems no reason to even if they won't break anything.

(Please test that regex, and don't just trust something I made up as I went!)

File: src/app/services/validation.ts [hunk 1]
```diff
@@ -19,6 +19,10 @@ export const validateEmail = (email?: string) => {
     return email && email.length > 0 && email.includes("@");
 };
 
+export const validateUrl = (url?: string) => {
```
Comment (jsharkey13) [inline src/app/services/validation.ts:22]: Sorry to be a further pain, but could this be `isValidGameboardId`?

Comment (jsharkey13) [reply src/app/services/validation.ts:22]: And take `gameboardId` as the argument?

Comment (jsharkey13) [reply src/app/services/validation.ts:22]: I think that's the correct way round, it returns true if valid, which is undefined or matching?

**Review by jsharkey13 — CHANGES_REQUESTED** [review]

File: src/app/services/validation.ts (hunk 1 above)
Comment (bh412) [reply src/app/services/validation.ts]: Done!

**Review by jsharkey13 — APPROVED** [review]

**Pull request merged**